plugins/module_utils/foreman_helper.py

```diff
@@ -499,10 +509,14 @@ def find_operatingsystems(self, names, **kwargs):
         return [self.find_operatingsystem(name, **kwargs) for name in names]
 
     def find_puppetclass(self, name, environment=None, params=None, failsafe=False, thin=None):
+        if thin is None:
+            thin = self._thin_default
         if environment:
             scope = {'environment_id': environment}
         else:
-            scope = None
+            scope = {}
+        if params is not None:
```

Comment:
funny, this was never used before? :)

Reply:
No, it wasn't. Also noone ever specified `thin` in the `entity_opts`. But you _can_ do that now...